### PR TITLE
Making book's purchase/ post method  (CLOSED PULL Request) 

### PR DESCRIPTION
**Pull request by AhmadAlkaati**
* Creating a responsive Login page .

* Setting inputs values in login page to session storage .

 *Displaying username in all project's pages when user is logged in.

* Adding a button for signing out beside username to all project's pages when a user is already logged in .

* When user clicks sign out button , username and password of the user are going to  be deleted from session storage and 
username and sign out button get replaced with login button.

* Showing buttons : 'Buy now' , 'Add to cart' and 'Add to wishlist' to only logged in user and hiding them when user is  not logged in.

* Adding social links to footers .

* working on responsivity of all pages .

NOTE : I'm sorry that this pull request has a lot of changes , I should have made more branches for  changes but I did all changes while I was working on last task and I forgot to do branches .

Thanks for reviewing in advance :)

### CODE REVIEW

File: books/book.js [hunk 1]
```diff
@@ -30,10 +30,75 @@ function displayBook(book) {
...
+      `${usernameSessionValue}:${passwordSessionValue}`
+    );
+    fetch("https://bookshop-api.mirkwood.dev/purchases/", {
+      method: "POST",
+      headers: {
+        Authorization: `Basic  ${base64EncodedString}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        id: 7,
```
Comment (josipgrgic) [inline books/book.js:89]: Id is generated by the backend, you do not need to specify it here

File: books/book.js [hunk 2]
```diff
@@ -30,10 +30,75 @@ function displayBook(book) {
...
+    const usernameSessionValue = sessionStorage.getItem("username");
+    const passwordSessionValue = sessionStorage.getItem("password");
+    // Authentication part //
+    const base64EncodedString = btoa(
+      `${usernameSessionValue}:${passwordSessionValue}`
+    );
+    fetch("https://bookshop-api.mirkwood.dev/purchases/", {
+      method: "POST",
+      headers: {
+        Authorization: `Basic  ${base64EncodedString}`,
```
Comment (josipgrgic) [inline books/book.js:85]: Wrap Authorization in quotes as well. So "Authorization"

Comment (AhmadAlkaati) [reply books/book.js:85]: I did that many times but quotes constantly get removed when I save.
I think it is caused by prettier extension.

Comment (josipgrgic) [reply books/book.js:85]: Ok, then. It's not a big issue

**Comment by josipgrgic** [comment]
Why does this PR contain login files (login.html, login.css, login.js)? That is Pouriya's job and only his PR should contain these files

File: books/book.js [hunk 3]
```diff
@@ -30,10 +30,75 @@ function displayBook(book) {
...
+    buttonsBuy.forEach((btn) => {
+      btn.style.display = "none";
+    });
+  }
+
+  signOutBtn.addEventListener("click", signOut);
+  function signOut() {
+    window.open("../index.html", "_self");
+    sessionStorage.removeItem("username");
+    sessionStorage.removeItem("password");
```
Comment (josipgrgic) [inline books/book.js:65]: I see this code in one form or in another in every js file. We generally don't want to repeat ourselves as programmers. I would suggest creating a navbar.js file in which you would define this behaviour and then just import that script to every html file

File: app.js [hunk 4]
```diff
@@ -26,3 +26,24 @@ function displayBooks(data) {
...
+  }
+  signOutBtn.addEventListener("click", signOut);
+  function signOut() {
+    window.open("index.html", "_self");
+    sessionStorage.removeItem("username");
+    sessionStorage.removeItem("password");
+  }
+}
+
+checkingUsernameSessionValue();
```
Comment (josipgrgic) [inline app.js:49]: Repeated code

File: books/book.js (hunk 1 above)
Comment (AhmadAlkaati) [reply books/book.js:89]: Alright

**Comment by AhmadAlkaati** [comment]
@josipgrgic I don't really understand what you mean by Pourija's job on login files.
But let me clarify things here : 
I didn't copy any of his code in login files, I did login html/css and javascript by myself. 
And I added more code in login.js file, as you know, purchase are happening in that file. 
And lastly I would download his code and work on it if it was merged into main but since it is not merged I worked on the latest merged code that did not include login files.

**Review by josipgrgic — CHANGES_REQUESTED** [review]
Everything seems to be working ok, but that is not enough.

The sign out logic is repeated. Imagine we have to change something in the future regarding signing out. We would have to change the same code in many different places. Please keep that i mind and always try to find ways how not to copy huge pieces of code to different places. We programmers have to be lazy in that regard :)

You need to **communicate** who does what. I'm sorry you did unnecessary work but you need to learn to work together and not overlap in your work. The point of this project is to learn working together as a team

You did mention you are aware of the fact there are many files but I need to tell again, that is not good practice. You shouldn't have unrelated work in a single PR. This PR has 700 new lines of code, that is a lot and only around 200 of these are truly needed to make a purchase, the other 500 are just noise. It's ok now, you can leave it as is. You will know better what I mean once you become the reviewer of someone else's code :D

File: books/book.js (hunk 3 above)
Comment (AhmadAlkaati) [reply books/book.js:65]: Alright. I will try to fix tomorrow

**Pull request closed without merging**